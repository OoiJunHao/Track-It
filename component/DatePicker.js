import React, { Component, useState } from "react";
import { TextInput, Text, StyleSheet, View } from 'react-native';
import { TouchableOpacity } from "react-native-gesture-handler";
import DateTimePickerModal from "react-native-modal-datetime-picker";
import Ionicons from "react-native-vector-icons/Ionicons";

const DatePicker = props => {
    const [isDatePickerVisible, setDatePickerVisibility] = useState(false);

    const showDatePicker = () => {
        setDatePickerVisibility(true);
    };

    const hideDatePicker = () => {
        setDatePickerVisibility(false);
        props.clear();
    };

    const handleConfirm = (date) => {
        console.log("A date has been picked: ", date);
        props.onChange(date);
        setDatePickerVisibility(false);
    };


    return (
        <View style={props.style}>
            <Text style={styles.inputTitle}>Expiry Date (Optional)</Text>
            <TouchableOpacity testID={'press'} style={styles.input} onPress={showDatePicker} >
                <Text testID={'date'} style={[styles.input, { marginTop: 10 }]}>{(props.date ? (new Date(props.date)).toDateString() : "")}</Text>
            </TouchableOpacity>
            <DateTimePickerModal
                testID={'picker'}
                isVisible={isDatePickerVisible}
                mode="date"
                cancelTextIOS="Clear"
                onConfirm={handleConfirm}
                onCancel={hideDatePicker}
            />
        </View>
    )
}

export default DatePicker


const styles = StyleSheet.create({
    inputTitle: {
        color: "#8A8F9E",
        fontSize: 10,
        textTransform: "uppercase"
    },
    input: {
        borderBottomColor: "#8A8F9E",
        borderBottomWidth: StyleSheet.hairlineWidth,
        height: 40,
        fontSize: 15,
        color: "#161f3d"
    },
    input2: {
        borderWidth: 2,
        borderRadius: 10,
        borderColor: 'rgb(121,130,167)',
        backgroundColor: "rgba(252,241,241,0.43)",
        textAlign: 'center',
        height: 40,
        fontSize: 15,
        color: "#161f3d"
    },
})