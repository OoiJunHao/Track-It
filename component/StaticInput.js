import React, { Component, useState } from "react";
import { TextInput, Text, StyleSheet, View } from 'react-native';
import { TouchableOpacity } from "react-native-gesture-handler";
import DateTimePickerModal from "react-native-modal-datetime-picker";
import Ionicons from "react-native-vector-icons/Ionicons";



export const StaticInput = props => (
    <View style={props.style} testID={'wrapper'}>
        <Text style={styles.inputTitle} testID={'text'}>{props.children}</Text>
        <TextInput style={styles.input} testID={'input'} autoCapitalize="none" onChangeText={props.onChangeText} value={props.value} />
    </View>
)

export const StaticInput2 = props => (
    <View style={props.style}>
        <Text style={styles.inputTitle}>{props.children}</Text>
        <TextInput style={styles.input2} autoCapitalize="none" keyboardType="email-address" onChangeText={props.onChangeText} value={props.value} placeholder="Type here" />
    </View>
)

export const StaticPasswordInput = props => (
    <View style={props.style}>
        <Text style={styles.inputTitle}>{props.children}</Text>
        <TextInput style={styles.input} autoCapitalize="none" secureTextEntry onChangeText={props.onChangeText} value={props.value} />
    </View>
)

export const StaticEmailInput = props => (
    <View style={props.style} testID={'wrapper'}>
        <Text style={styles.inputTitle} testID={'text'}>{props.children}</Text>
        <TextInput style={styles.input} testID={'input'} autoCapitalize="none" keyboardType="email-address" onChangeText={props.onChangeText} value={props.value} />
    </View>
)

export const FieldInput = props => (
    <View style={props.style}>
        <Text style={styles.inputTitle}>{props.children}</Text>
        <TextInput style={styles.input} onChangeText={props.onChangeText} value={props.value} />
    </View>
)

export const DatePicker = props => {
    const [isDatePickerVisible, setDatePickerVisibility] = useState(false);

    const showDatePicker = () => {
        setDatePickerVisibility(true);
    };

    const hideDatePicker = () => {
        setDatePickerVisibility(false);
        props.clear();
    };

    const handleConfirm = (date) => {
        console.log("A date has been picked: ", date);
        props.onChange(date);
        setDatePickerVisibility(false);
    };


    return (
        <View style={props.style}>
            <Text style={styles.inputTitle}>Expiry Date (Optional)</Text>
            <TouchableOpacity style={styles.input} onPress={showDatePicker}>
                <Text style={[styles.input, { marginTop: 10 }]}>{(props.date ? (new Date(props.date)).toDateString() : "")}</Text>
            </TouchableOpacity>
            <DateTimePickerModal
                isVisible={isDatePickerVisible}
                mode="date"
                cancelTextIOS="Clear"
                onConfirm={handleConfirm}
                onCancel={hideDatePicker}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    inputTitle: {
        color: "#8A8F9E",
        fontSize: 10,
        textTransform: "uppercase"
    },
    input: {
        borderBottomColor: "#8A8F9E",
        borderBottomWidth: StyleSheet.hairlineWidth,
        height: 40,
        fontSize: 15,
        color: "#161f3d"
    },
    input2: {
        borderWidth: 2,
        borderRadius: 10,
        borderColor: 'rgb(121,130,167)',
        backgroundColor: "rgba(252,241,241,0.43)",
        textAlign: 'center',
        height: 40,
        fontSize: 15,
        color: "#161f3d"
    },
})
