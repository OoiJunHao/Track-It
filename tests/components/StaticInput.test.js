import React from 'react';
import StaticInput from "../../component/Input"
import StaticPasswordInput from "../../component/StaticPasswordInput"
import StaticEmailInput from "../../component/StaticEmail"
import { render, cleanup, fireEvent, toBeTruthy } from 'react-native-testing-library';
import DatePicker from '../../component/DatePicker';

afterEach(cleanup);

describe('<StaticInput />', () => {
    it('should match snapshot', () => {
        const rendered = render(<StaticInput children={'abacaba'} />).toJSON();
        expect(rendered).toMatchSnapshot();
    })

    it('should properly render the text', () => {
        const rendered = render(<StaticInput children={'abacaba'} />);
        const textComponent = rendered.getByTestId('text');
        expect(textComponent.props.children).toEqual('abacaba');
    });

    it('should fire onChange events', () => {
        const onChange = jest.fn();
        const rendered = render(<StaticInput onChangeText={onChange} />);
        const inputComponent = rendered.getByTestId('input');

        fireEvent(inputComponent, 'changeText', 'new text');
        expect(onChange).toHaveBeenCalledWith('new text');
    });
})

describe('<StaticPasswordInput />', () => {
    it('should match snapshot', () => {
        const rendered = render(<StaticPasswordInput children={'Password'} />).toJSON();
        expect(rendered).toMatchSnapshot();
    })

    it('should properly render the text', () => {
        const rendered = render(<StaticPasswordInput children={'Password'} />);
        const textComponent = rendered.getByTestId('text');
        expect(textComponent.props.children).toEqual('Password');
    });

    it('should fire onChange events', () => {
        const onChange = jest.fn();
        const rendered = render(<StaticPasswordInput onChangeText={onChange} />);
        const inputComponent = rendered.getByTestId('input');
        fireEvent(inputComponent, 'changeText', 'new text');
        expect(onChange).toHaveBeenCalledWith('new text');
    });

    /* it('password appears as asterisk or bulletpoints', () => {
        const onChange = jest.fn();
        const rendered = render(<StaticPasswordInput onChangeText={onChange} />);
        const inputComponent = rendered.getByTestId('input');
        expect(inputComponent.children.secureTextEntry).toBeTruthy();
    }) */
})

describe('<StaticEmailInput />', () => {
    it('should match snapshot', () => {
        const rendered = render(<StaticEmailInput children={'abacaba'} />).toJSON();
        expect(rendered).toMatchSnapshot();
    })

    it('should properly render the text', () => {
        const rendered = render(<StaticEmailInput children={'abacaba'} />);
        const textComponent = rendered.getByTestId('text');
        expect(textComponent.props.children).toEqual('abacaba');
    });

    it('should fire onChange events', () => {
        const onChange = jest.fn();
        const rendered = render(<StaticEmailInput onChangeText={onChange} />);
        const inputComponent = rendered.getByTestId('input');

        fireEvent(inputComponent, 'changeText', 'new text');
        expect(onChange).toHaveBeenCalledWith('new text');
    });
})


describe('<DatePicker />', () => {
    it('should match snapshot', () => {
        const rendered = render(<DatePicker />).toJSON();
        expect(rendered).toMatchSnapshot();
    })

    it('should properly render the text', () => {
        const rendered = render(<DatePicker />);
        const textComponent = rendered.getByTestId('date');
        expect(textComponent.props.children).toEqual("");
    });

    it('should open date picker', () => {
        const isVisible = jest.fn();
        const rendered = render(<DatePicker />);
        const inputComponent = rendered.getByTestId('press');

        fireEvent.press(inputComponent);
    });
})



