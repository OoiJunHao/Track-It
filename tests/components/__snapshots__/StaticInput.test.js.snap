// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> should match snapshot 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#8A8F9E",
        "fontSize": 10,
        "textTransform": "uppercase",
      }
    }
  >
    Expiry Date (Optional)
  </Text>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "borderBottomColor": "#8A8F9E",
          "borderBottomWidth": 0.5,
          "color": "#161f3d",
          "fontSize": 15,
          "height": 40,
          "opacity": 1,
        }
      }
      testID="press"
    >
      <Text
        style={
          Array [
            Object {
              "borderBottomColor": "#8A8F9E",
              "borderBottomWidth": 0.5,
              "color": "#161f3d",
              "fontSize": 15,
              "height": 40,
            },
            Object {
              "marginTop": 10,
            },
          ]
        }
        testID="date"
      >
        
      </Text>
    </View>
  </RNGestureHandlerButton>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    isVisible={false}
    onHide={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
  </Modal>
</View>
`;

exports[`<StaticEmailInput /> should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <Text
    style={
      Object {
        "color": "#8A8F9E",
        "fontSize": 10,
        "textTransform": "uppercase",
      }
    }
    testID="text"
  >
    abacaba
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    keyboardType="email-address"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomColor": "#8A8F9E",
        "borderBottomWidth": 0.5,
        "color": "#161f3d",
        "fontSize": 15,
        "height": 40,
      }
    }
    testID="input"
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`<StaticInput /> should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <Text
    style={
      Object {
        "color": "#8A8F9E",
        "fontSize": 10,
        "textTransform": "uppercase",
      }
    }
    testID="text"
  >
    abacaba
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomColor": "#8A8F9E",
        "borderBottomWidth": 0.5,
        "color": "#161f3d",
        "fontSize": 15,
        "height": 40,
      }
    }
    testID="input"
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`<StaticPasswordInput /> should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <Text
    style={
      Object {
        "color": "#8A8F9E",
        "fontSize": 10,
        "textTransform": "uppercase",
      }
    }
    testID="text"
  >
    Password
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "borderBottomColor": "#8A8F9E",
        "borderBottomWidth": 0.5,
        "color": "#161f3d",
        "fontSize": 15,
        "height": 40,
      }
    }
    testID="input"
    underlineColorAndroid="transparent"
  />
</View>
`;
